apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/data_updater/impl.ex

```diff
@@ -1207,7 +1216,44 @@ defmodule Astarte.DataUpdaterPlant.DataUpdater.Impl do
     :ok = Queries.add_old_interfaces(db_client, new_state.device_id, old_introspection)
     :ok = Queries.remove_old_interfaces(db_client, new_state.device_id, readded_introspection)
 
-    # TODO: handle triggers for interface minor updates
+    interface_minor_updated_map =
+      Enum.map(old_minors, fn {iface, old_minor} ->
+        old_major = Map.fetch!(state.introspection, iface)
+        {iface, old_major, old_minor}
+      end)
+      |> Enum.filter(fn {iface, old_major, _old_minor} ->
+        Enum.member?(db_introspection_map, {iface, old_major})
+      end)
+      |> Enum.reduce(%{}, fn {iface, old_major, old_minor}, acc ->
+        new_minor = Map.get(db_introspection_minor_map, iface)
+
+        if new_minor > old_minor do
```

Comment:
The trigger is a generic "updated", so shouldn't this be `new_minor != old_minor`? Technically, while rare, a device could downgrade the minor version of an interface (e.g. software downgrade reverting to an older interface version)